Path: Rename Path-*.svg files to conform with the FC image file naming standard

luzpaz opened the pull request:
This PR brings the Path WB in to uniformity with the rest (majority) of the FreeCAD workbench codebase. This also aligns the Path workbench with the wiki documentation. Commits have been split up for easier review.

Forum thread: https://forum.freecadweb.org/viewtopic.php?f=21&t=52715#p452835

luzpaz commented on src/Mod/Path/PathScripts/PathAdaptiveGui.py:
```diff
@@ -214,6 +214,10 @@ def getFields(self, obj):
+Command = PathOpGui.SetupOperation('Adaptive',
+        PathAdaptive.Create,
+        TaskPanelOpPage,
+        'Path_Adaptive',
```
I assumed that line 219 reflected the name of the icon, is that correct?

chennes replied:
Yes, that's right.

luzpaz commented on src/Mod/Path/PathScripts/PathAdaptiveGui.py:
```diff
@@ -214,6 +214,10 @@ def getFields(self, obj):
+        TaskPanelOpPage,
+        'Path_Adaptive',
+        QtCore.QT_TRANSLATE_NOOP("Path_Adaptive", "Adaptive"),
+        QtCore.QT_TRANSLATE_NOOP("Path_Pocket", "Adaptive clearing and profiling")
```
The translation context is `Path_Pocket` but it looks like it should be `Path_Adaptive`. If so, this was a pre-existing mistake in the code. Shall I fix?

chennes replied:
I think you are correct (comparing, e.g. to PathCustomGui.py lines 80 and 81), but changing this will require changing all of the .ts files, too, as far as I know. Then again, the UK .ts entry for the adaptive clearing and profiling looks very suspect!

luzpaz replied:
There is another pending PR (#4093 ) that will change all the .ts files, so no worries. This should be fixed since it helps the translators understand the context of the translation. I'll make the change now.

luzpaz replied:
Fixed in 26a434b

luzpaz commented on src/Mod/Path/PathScripts/PathDeburrGui.py:
```diff
@@ -135,7 +135,7 @@ def taskPanelBaseGeometryPage(self, obj, features):
         PathDeburr.Create,
         TaskPanelOpPage,
-        'Path-Deburr',
+        'Path_Deburr',
```
I made the same assumption (as mentioned above) for all `PathOpGui.SetupOperation()` functions

luzpaz commented on src/Mod/Path/PathScripts/post/gcode_pre.py:
```diff
@@ -112,7 +112,11 @@ def insert(filename, docname):
         # Create a custom and viewobject
         obj = PathCustom.Create("Custom")
-        res = PathOpGui.CommandResources('Custom', PathCustom.Create, PathCustomGui.TaskPanelOpPage, 'Path-Custom', 'Path-Custom', '', '')
+        res = PathOpGui.CommandResources('Custom',
+            PathCustom.Create, PathCustomGui.TaskPanelOpPage,
+            'Path_Custom', 
+            'Path_Custom', '', ''
+            )
```
I modified this. Does it look correct or shall I revert?

chennes replied:
The fourth parameter there is the pixmap, but the fifth is the menu text. I doubt it is supposed to have either an underscore or a dash in it, I'm betting there is some missing translation code -- something like
```suggestion
        res = PathOpGui.CommandResources('Custom',
            PathCustom.Create, PathCustomGui.TaskPanelOpPage,
            'Path_Custom', 
            QtCore.QT_TRANSLATE_NOOP('Path_Custom', 'Custom'), '', ''
            )
```

luzpaz replied:
@mlampert care to weigh in?

Russ4262 replied:
Yes, Dr. Hennes.  That fifth item should include a translation call, like:
`QtCore.QT_TRANSLATE_NOOP("PathCustom", "Custom Path")` as you included in your suggestion.  I believe (but have not tracked down source code function) the first argument in the translate function used is a general location reference, like the module file name, hence "PathCustom", "PathHelix", or "PathPocket."

chennes replied:
lol, nobody calls me "Dr. Hennes" unless I've just screwed something up and they're about to point it out :). You are right about the first argument to [QT_TRANSLATE_NOOP](https://doc.qt.io/qt-5/i18n-source-translation.html#using-qt-tr-noop-and-qt-translate-noop-in-c): Qt calls it the "Context," it's used to group together various translation items in the .ts file.

luzpaz replied:
> `QtCore.QT_TRANSLATE_NOOP("PathCustom", "Custom Path")` as you included in your suggestion. I believe (but have not tracked down source code function) the first argument in the translate function used is a general location reference, like the module file name, hence "PathCustom", "PathHelix", or "PathPocket."

@Russ4262 thanks for bringing this up. I do have a commit that I didn't add to this PR that changes all the translation context fields to have an underscore (for several reasons, because there is a mix in the Path code of camelcase and underscore and also because of the wiki documentation/images have underscores). But I didn't include it because it would create a [kitten eating moster](http://webchick.net/please-stop-eating-baby-kittens) PR. It would also render all/many Path translations obsolete pre-release. So I'm waiting till after 0.19 release to submit it

luzpaz reviewed (commented):
Sorry for this PITA PR. We will have more eyes on the code and testers now since we are close to a potential release and so now is a good time IMHO to implement this.

chennes commented:
I see that you replaced the first underscore in each filename, but left later ones (e.g. "Path_Area-Workplane.svg"). My understanding of the reason for this change to underscores was that the dash was problematic on the wiki -- is there a reason that the later dash is OK, but the earlier is not?

luzpaz commented:
> I see that you replaced the first underscore in each filename, but left later ones (e.g. "Path_Area-Workplane.svg"). My understanding of the reason for this change to underscores was that the dash was problematic on the wiki -- is there a reason that the later dash is OK, but the earlier is not?

That was an oversight on my part, @chennes :wink:  
Fixed in 13042af and 2550def

luzpaz commented:
This PR should be ready to go

chennes reviewed (approved):
This compiles and runs, and as far as I can tell all of the Path icons still show up correctly. It seems there are a few that aren't being used, or at least I can't see where they would appear in the interface (e.g. Path_Kurve.svg) -- I couldn't really test those for anything.

luzpaz commented:
@Russ4262 do you mind reviewing this PR?

sliptonic commented:
I agree with renaming files to conform to wiki.  Basically I agree with the intent and execution of this PR but still have some concerns.  Since it touches nearly every file in the Path workbench, it will likely force rebasing of all outstanding PRs.  Even if this PR doesn't introduce any regressions, rebasing all the other PRs might.  
Is this an absolutely necessary and wise thing to do this close to a release?  (I admit I have no clue how close to a release we actually are)

Russ4262 commented:
I will try for this evening. Looks like Sliptonic and Chennes have great feedback.

luzpaz commented:
@sliptonic I don't want to step on anybodies toes here so for the purpose of being time sensitive and practice I'm just going to write things casually but please understand that my intention is to help move multiple aspects of FC forward, documentation being one of them, so they keep up with the source code development.

> Since it touches nearly every file in the Path workbench, it will likely force rebasing of all outstanding PRs. Even if this PR doesn't introduce any regressions, rebasing all the other PRs might.

IMHO, it would merit testing it out. Just a cursory search through other open Path PRs returns the following impressions for me (i'm no expert but lets hypothesize)

#4097 doesn't look like it would be affected  
#4096 (your PR) doesn't look affected
#4026 doesn't look like it would be affected
#3935 doesn't look like it would be affected

#4089 blocks this PR
#4071 probably blocks this PR 

Essentially, 2 PRs are potentially blocking this one.

Observation: 2 commits in this PR you can easily ignore (but take up a good portion of the touched files) because they change the metadata of the SVGs. An aside, the SVGs really need to reformatted so they are saved without inkscape formatting in them (but that's a separate issue that can wait for another time) 
059c932 (reviewers can skip reviewing this)
5f9f595 (reviewers can skip reviewing this)

> Is this an absolutely necessary and wise thing to do this close to a release? (I admit I have no clue how close to a release we actually are)

Well lets say hypothetically, we have a month till release, and that's being conservative since no feature freezes have been announced etc...then I think this is enough time to test all the commands. It's also enough time to * potentially * get the Path documentation in order for Path WB.

Russ4262 commented:
@luzpaz 
I think the PR looks good as well.  The only thing I might suggest is to go ahead and add the fix for the missing translation function in the PathCustomGui module.  

Great job.  Thanks for making FreeCAD better!

luzpaz commented:
@Russ4262 I've added @chennes's commit. I also added `[skip ci]` to it so it won't trigger Travis. I've also rebased the commit. Unless you want to wait for the other 2 pending PRs that I mentioned in https://github.com/FreeCAD/FreeCAD/pull/4095#issuecomment-736707638, this PR is good to go, IMHO.

luzpaz commented:
Needed to resort `Path.qrc` file again alphabetically (27b71b1). Now everything is in order
